src/common/fetchClient.ts

```diff
@@ -0,0 +1,154 @@
+export interface FetchClientConfig {
+  baseURL?: string;
+  headers?: Record<string, string>;
+}
+
+export interface FetchRequestConfig<D = any> {
+  params?: Record<string, string>;
+  headers?: Record<string, string>;
+  body?: D;
+}
+
+export interface FetchResponse<T = any> extends Response {
+  data: T;
+}
+
+export interface FetchResponseErrorData<T = any> {
+  status: number;
+  headers: Record<string, string>;
+  data: T;
+}
+
+export class FetchResponseError extends Error {
+  readonly response: FetchResponseErrorData;
+
+  constructor(response: FetchResponseErrorData) {
```

Comment:
```suggestion
  constructor(response: FetchResponse) {
```
Continuing on from comment/question re: `this.headersToObject()` below - if we can skip and just use `headers.get()`, then I'm guessing we can use `FetchResponse` here without needing to define a new `FetchResponseErrorData` type?